### ISSUE TITLE
Fatal Exception: NSInvalidArgumentException -[__NSDictionaryM encodeSuccessEnvelope:]: unrecognized selector sent to instance

### ISSUE THREAD

**Issue by Tom3652** [flutter/flutter#95565]
## Steps to Reproduce

Not sure how to reproduce this but i have more information than usual on this kind of errors. 
The error name is the following : 

<img width="393" alt="Capture d’écran 2021-12-20 à 15 07 25" src="https://user-images.githubusercontent.com/32631467/146779657-48cd141e-6b95-48e9-af4a-563d47f42560.png">

And the description is : 

```
Fatal Exception: NSInvalidArgumentException -[__NSDictionaryM encodeSuccessEnvelope:]: unrecognized selector sent to instance
```

The ```keys``` : 

<img width="1154" alt="Capture d’écran 2021-12-20 à 15 08 16" src="https://user-images.githubusercontent.com/32631467/146779751-69bd51a9-4aa6-43c2-af89-7d6e6f3222b3.png">

The ```logs``` :

<img width="881" alt="Capture d’écran 2021-12-20 à 15 08 52" src="https://user-images.githubusercontent.com/32631467/146779824-c674b40d-c615-4367-9f1c-4270543f8649.png">

I actually see ```photo_manager``` in the logs. Could this be an issue from this [package](https://pub.dev/packages/photo_manager) rather than flutter itself ? 

I am using this package to load phone's asset in a ```Gridview.builder```.

**Expected results:** 

The app should not crash.

**Actual results:** 

The app is crashing.

<details>
  <summary>Logs</summary>

```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x9904c __exceptionPreprocess
1  libobjc.A.dylib                0x15f54 objc_exception_throw
2  CoreFoundation                 0x176014 +[NSObject(NSObject) _copyDescription]
3  CoreFoundation                 0x2e474 ___forwarding___
4  CoreFoundation                 0x2d5b0 _CF_forwarding_prep_0
5  Flutter                        0x4f02b0 (Missing)
6  Flutter                        0x3c664 (Missing)
7  Flutter                        0x3bc204 (Missing)
8  Flutter                        0x2eb0d8 (Missing)
9  Flutter                        0x2ee6a4 (Missing)
10 CoreFoundation                 0xae268 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
11 CoreFoundation                 0x32c40 __CFRunLoopDoTimer
12 CoreFoundation                 0x2d43c __CFRunLoopDoTimers
13 CoreFoundation                 0xbc58 __CFRunLoopRun
14 CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific
15 GraphicsServices               0x138c GSEventRunModal
16 UIKitCore                      0x5196a8 -[UIApplication _run]
17 UIKitCore                      0x2987f4 UIApplicationMain
18 Runner                         0x7ab0 main + 7 (AppDelegate.swift:7)
19 ???                            0x10702da24 (Missing)

Crashed: com.google.firebase.crashlytics.ios.exception
0  FirebaseCrashlytics            0x1f44c FIRCLSProcessRecordAllThreads + 184
1  FirebaseCrashlytics            0x1f82c FIRCLSProcessRecordAllThreads + 1176
2  FirebaseCrashlytics            0x167f0 FIRCLSHandler + 48
3  FirebaseCrashlytics            0x10c78 __FIRCLSExceptionRecord_block_invoke + 92
4  libdispatch.dylib              0x4660 _dispatch_client_callout + 20
5  libdispatch.dylib              0x138b4 _dispatch_lane_barrier_sync_invoke_and_complete + 56
6  FirebaseCrashlytics            0x10414 FIRCLSExceptionRecord + 216
7  FirebaseCrashlytics            0x10720 FIRCLSExceptionRecordNSException + 560
8  FirebaseCrashlytics            0xffec FIRCLSTerminateHandler() + 404
9  libc++abi.dylib                0x12f58 std::__terminate(void (*)()) + 20
10 libc++abi.dylib                0x15efc __cxa_rethrow + 148
11 libobjc.A.dylib                0x17e8c objc_exception_rethrow + 44
12 CoreFoundation                 0x1f4a4 CFRunLoopRunSpecific + 836
13 GraphicsServices               0x138c GSEventRunModal + 164
14 UIKitCore                      0x5196a8 -[UIApplication _run] + 1100
15 UIKitCore                      0x2987f4 UIApplicationMain + 2092
16 Runner                         0x7ab0 main + 7 (AppDelegate.swift:7)
17 ???                            0x10702da24 (Missing)

com.apple.coremedia.player.async.0x280300fc0.P/TZ
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  libdispatch.dylib              0x1f27c _dispatch_mach_send_and_wait_for_reply + 520
3  libdispatch.dylib              0x1f62c dispatch_mach_send_with_result_and_wait_for_reply + 56
4  libxpc.dylib                   0xeb9c xpc_connection_send_message_with_reply_sync + 240
5  CoreMedia                      0xed5d0 FigXPCRemoteClientSendSyncMessageCreatingReply + 44
6  MediaToolbox                   0x25592c remoteXPCPlayer_SetRateWithOptions + 152
7  MediaToolbox                   0x13dd58 playerasync_runOneCommand + 4844
8  MediaToolbox                   0x1412e4 playerasync_runAsynchronousCommandOnQueue + 184
9  libdispatch.dylib              0x4660 _dispatch_client_callout + 20
10 libdispatch.dylib              0xbde4 _dispatch_lane_serial_drain + 672
11 libdispatch.dylib              0xc958 _dispatch_lane_invoke + 392
12 libdispatch.dylib              0x171a8 _dispatch_workloop_worker_thread + 656
13 libsystem_pthread.dylib        0x10f4 _pthread_wqthread + 288
14 libsystem_pthread.dylib        0xe94 start_wqthread + 8

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Foundation                     0x18354 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 236
6  Foundation                     0x59c28 -[NSRunLoop(NSRunLoop) runUntilDate:] + 92
7  UIKitCore                      0x4928a4 -[UIEventFetcher threadMain] + 524
8  Foundation                     0x6836c __NSThread__start__ + 808
9  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
10 libsystem_pthread.dylib        0xea0 thread_start + 8

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  CFNetwork                      0x27a6a0 _CFURLStorageSessionCopyIdentifier + 59784
6  Foundation                     0x6836c __NSThread__start__ + 808
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

io.flutter.1.ui
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Flutter                        0x2ee570 (Missing)
6  Flutter                        0x2edd30 (Missing)
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

io.flutter.1.raster
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Flutter                        0x2ee570 (Missing)
6  Flutter                        0x2edd30 (Missing)
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

io.flutter.1.io
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Flutter                        0x2ee570 (Missing)
6  Flutter                        0x2edd30 (Missing)
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.1
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.2
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.3
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.4
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.5
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.6
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

dart:io EventHandler
0  libsystem_kernel.dylib         0x2e7c kevent + 8
1  Flutter                        0x4b14dc (Missing)
2  Flutter                        0x4e04e8 (Missing)
3  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
4  libsystem_pthread.dylib        0xea0 thread_start + 8

MRAVConcreteRoutingDiscoverySession.reloadQueue
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  libdispatch.dylib              0x1f27c _dispatch_mach_send_and_wait_for_reply + 520
3  libdispatch.dylib              0x1f62c dispatch_mach_send_with_result_and_wait_for_reply + 56
4  libxpc.dylib                   0xeb9c xpc_connection_send_message_with_reply_sync + 240
5  CoreMedia                      0xed5d0 FigXPCRemoteClientSendSyncMessageCreatingReply + 44
6  CoreMedia                      0xed544 FigXPCSendStdCopyPropertyMessage + 112
7  MediaExperience                0x173d0 FigRouteDiscovererRemoteXPC_CopyProperty + 120
8  AVFCore                        0x2c924 -[AVFigRouteDiscovererOutputDeviceDiscoverySessionImpl availableOutputDevicesObject] + 284
9  AVFCore                        0x2af2c -[AVOutputDeviceDiscoverySession availableOutputDevices] + 64
10 MediaRemote                    0x3afe0 -[MRAVConcreteRoutingDiscoverySession _onQueue_reloadAvailableOutputDevices] + 168
11 MediaRemote                    0x4dd54 __76-[MRAVConcreteRoutingDiscoverySession _scheduleAvailableOutputDevicesReload]_block_invoke_2 + 136
12 libdispatch.dylib              0x2914 _dispatch_call_block_and_release + 32
13 libdispatch.dylib              0x4660 _dispatch_client_callout + 20
14 libdispatch.dylib              0xbde4 _dispatch_lane_serial_drain + 672
15 libdispatch.dylib              0xc958 _dispatch_lane_invoke + 392
16 libdispatch.dylib              0x171a8 _dispatch_workloop_worker_thread + 656
17 libsystem_pthread.dylib        0x10f4 _pthread_wqthread + 288
18 libsystem_pthread.dylib        0xe94 start_wqthread + 8

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

com.google.firebase.crashlytics.MachExceptionServer
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  FirebaseCrashlytics            0x19140 FIRCLSMachExceptionServer + 104
3  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
4  libsystem_pthread.dylib        0xea0 thread_start + 8

AVAudioSession Notify Thread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  AudioSession                   0x6308 CADeprecated::GenericRunLoopThread::Entry(void*) + 164
6  AudioSession                   0xfd64 CADeprecated::CAPThread::Entry(CADeprecated::CAPThread*) + 92
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

XPCConnectionManager:0x280d345b0
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  libdispatch.dylib              0x1f27c _dispatch_mach_send_and_wait_for_reply + 520
3  libdispatch.dylib              0x1f62c dispatch_mach_send_with_result_and_wait_for_reply + 56
4  libxpc.dylib                   0xeb9c xpc_connection_send_message_with_reply_sync + 240
5  Foundation                     0x3f0a4 __NSXPCCONNECTION_IS_WAITING_FOR_A_SYNCHRONOUS_REPLY__ + 16
6  Foundation                     0x44f44 -[NSXPCConnection _sendInvocation:orArguments:count:methodSignature:selector:withProxy:] + 2508
7  Foundation                     0x7e124 -[NSXPCConnection _sendSelector:withProxy:arg1:arg2:] + 144
8  Foundation                     0x92ce8 _NSXPCDistantObjectSimpleMessageSend2 + 76
9  CoreData                       0xb8000 -[NSXPCStoreConnection sendMessage:store:error:] + 372
10 CoreData                       0x12566c -[NSXPCStoreConnection sendMessageWithContext:] + 48
11 CoreData                       0xb73f4 __54-[NSXPCStoreConnectionManager sendMessageWithContext:]_block_invoke + 92
12 libdispatch.dylib              0x4660 _dispatch_client_callout + 20
13 libdispatch.dylib              0x13f08 _dispatch_sync_invoke_and_complete + 56
14 CoreData                       0x1158d0 -[NSXPCStoreConnectionManager sendMessageWithContext:] + 112
15 CoreData                       0xae8cc -[NSXPCStore sendMessage:fromContext:interrupts:error:] + 372
16 CoreData                       0x93b98 -[NSXPCStore executeFetchRequest:withContext:error:] + 700
17 CoreData                       0x1062a8 -[NSXPCStore executeRequest:withContext:error:] + 148
18 CoreData                       0x1c79a4 __65-[NSPersistentStoreCoordinator executeRequest:withContext:error:]_block_invoke.819 + 2436
19 CoreData                       0x2a71c -[NSPersistentStoreCoordinator _routeHeavyweightBlock:] + 264
20 CoreData                       0x5bdf8 -[NSPersistentStoreCoordinator executeRequest:withContext:error:] + 1248
21 CoreData                       0x20dc4 -[NSManagedObjectContext executeFetchRequest:error:] + 860
22 PhotoLibraryServices           0xb390 -[PLManagedObjectContext pl_executeFetchRequest:error:] + 280
23 PhotoLibraryServices           0x27eb0 -[PLManagedObjectContext executeFetchRequest:error:] + 108
24 Photos                         0x287c4 __55+[PHFetchResult fetchObjectIDs:inManagedObjectContext:]_block_invoke + 224
25 PhotoLibraryServices           0x1f0528 __46-[PLManagedObjectContext performBlockAndWait:]_block_invoke + 96
26 CoreData                       0x1f754 developerSubmittedBlockToNSManagedObjectContextPerform + 156
27 libdispatch.dylib              0x4660 _dispatch_client_callout + 20
28 libdispatch.dylib              0x138b4 _dispatch_lane_barrier_sync_invoke_and_complete + 56
29 CoreData                       0x1f9b4 -[NSManagedObjectContext performBlockAndWait:] + 264
30 PhotoLibraryServices           0x1f04a8 -[PLManagedObjectContext performBlockAndWait:] + 148
31 Photos                         0x16064 +[PHFetchResult fetchObjectIDs:inManagedObjectContext:] + 428
32 Photos                         0x12140 __57-[PHFetchResult fetchedObjectsUsingManagedObjectContext:]_block_invoke + 196
33 PhotoLibraryServicesCore       0x8560 __pl_dispatch_sync_block_invoke + 44
34 libdispatch.dylib              0x4660 _dispatch_client_callout + 20
35 libdispatch.dylib              0x138b4 _dispatch_lane_barrier_sync_invoke_and_complete + 56
36 PhotoLibraryServicesCore       0x86c0 pl_dispatch_sync + 124
37 Photos                         0x24ad0 -[PHFetchResult fetchedObjectsUsingManagedObjectContext:] + 132
38 Photos                         0x1e294 -[PHFetchResult initWithQuery:oids:registerIfNeeded:usingManagedObjectContext:] + 716
39 Photos                         0x1afc8 -[PHFetchResult initWithQuery:] + 96
40 Photos                         0x21d8c __23-[PHQuery executeQuery]_block_invoke + 48
41 Photos                         0x48e68 PHQueryForAssetInAlbumKind_UserLibraryAlbum_block_invoke_50 + 212
42 Photos                         0x13a98 -[PHQuery executeQuery] + 124
43 Photos                         0x1fa8c +[PHAsset fetchAssetsInAssetCollection:options:] + 628
44 photo_manager                  0x9ac0 -[PMManager getAssetEntityListWithRange:type:start:end:filterOption:] + 364
45 photo_manager                  0x133e4 __26-[PMPlugin onAuth:result:]_block_invoke + 1972
46 libdispatch.dylib              0x2914 _dispatch_call_block_and_release + 32
47 libdispatch.dylib              0x4660 _dispatch_client_callout + 20
48 libdispatch.dylib              0x7788 _dispatch_queue_override_invoke + 792
49 libdispatch.dylib              0x15dd0 _dispatch_root_queue_drain + 396
50 libdispatch.dylib              0x165f8 _dispatch_worker_thread2 + 164
51 libsystem_pthread.dylib        0x10b8 _pthread_wqthread + 228
52 libsystem_pthread.dylib        0xe94 start_wqthread + 8

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

com.apple.CoreMotion.MotionThread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  CoreFoundation                 0x9fa78 CFRunLoopRun + 64
6  CoreMotion                     0x13544 CLMotionActivity::isTypeInVehicle(CLMotionActivity::Type) + 21264
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

com.squareup.SocketRocket.NetworkThread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Foundation                     0x18354 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 236
6  FirebaseDatabase               0x52f5c -[_FSRRunLoopThread main] + 252
7  Foundation                     0x6836c __NSThread__start__ + 808
8  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
9  libsystem_pthread.dylib        0xea0 thread_start + 8

com.apple.CFSocket.private
0  libsystem_kernel.dylib         0x21d0 select$DARWIN_EXTSN + 8
1  libsystem_kernel.dylib         0x21d0 __select + 8
2  CoreFoundation                 0xaa740 __CFSocketManager + 648
3  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
4  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @Tom3652, please provide the output of `flutter doctor -v`. 

Thank you

**Comment by Tom3652** [comment]
Yes sure sorry i forgot it : 

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale fr-FR)
    • Flutter version 2.8.1 at /Users/foxtom/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (5 days ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/foxtom/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.63.2)
    • VS Code at /Users/foxtom/Desktop/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 96.0.4664.110

• No issues found!
```

**Comment by danagbemava-nc** [comment]
@flutter-symbolizer-bot https://github.com/flutter/flutter/issues/95565#issue-1084850159 force flutter#v2.8.0 ios release arm64

**Comment by danagbemava-nc** [comment]
@flutter-symbolizer-bot https://github.com/flutter/flutter/issues/95565#issue-1084850159 force flutter#v2.8.1 ios release arm64

**Comment by jmagman** [comment]
`encodeSuccessEnvelope` should be called on a `FlutterMethodCodec`
https://github.com/flutter/engine/blob/d91489e9f62fcdab9d43783863c8c46b1e33b236/shell/platform/darwin/common/framework/Source/FlutterChannels.mm#L308

Which I believe is passed in from:
https://github.com/flutter/engine/blob/d91489e9f62fcdab9d43783863c8c46b1e33b236/shell/platform/darwin/common/framework/Source/FlutterChannels.mm#L279-L280
```objc
  NSObject<FlutterMethodCodec>* codec = [FlutterStandardMethodCodec sharedInstance];
  return [FlutterEventChannel eventChannelWithName:name binaryMessenger:messenger codec:codec];
```

So how is that ever a dictionary?

**Comment by gaaclarke** [comment]
A few things are happening here, this is a bit confusing.  So, objc is throwing an exception on the platform thread, the exception is the unrecognized selector error.  The developer is using crashlytics that has registered itself with std::terminate to get called in such an event.  That handler is crashing on thread `com.google.firebase.crashlytics.ios.exception`.  I'm unsure how photo_manager is related, it could be the plugin causing the unrecognized selector error or it could be some other plugin.  It's also unclear how "exception: invalid image data" is related to this crash.  Especially since the unrecognized selector is `encodeSuccessEnvelope` which would mean something went right, not wrong.

So, `encodeSuccessEnvelope` is a method on FlutterCodec, not NSDictionary like jenn pointed out.  So there are a few reason that could happen.  The most likely is that that address in memory used to point to a FlutterCodec, but wasn't managed correctly so it was deleted and later assigned to an NSDictionary.

@zanderso what safeguards are there that a plugin behaves?  None.  The plugins don't execute in some sort of sandbox and there is no automated process that verifies that plugins meet some level of quality.

So:
1) Crashlytics shouldn't crash, that's a bug that should be filed with them.
1) You are probably right that it is `photo_manager` that is misbehaving but unfortunately the problem can't be definitively attributed to them or to the engine without more information, preferably reproduction code for the bug.

**Comment by Tom3652** [comment]
Thanks for the detailed explanation.

I am sorry i can't reproduce it as it is one of my beta testers that had this crash and I have never encountered it during development or profile / release mode myself... 

Maybe it is fixed in a newer ```photo_manager``` version, I have upgraded it today to ```2.0.0.dev-02```, i was using the ```1.3.10``` at the moment of this crash. 

However, If this was a plugin issue, I think i would encounter it more often because the app is used quite a lot every day and on various devices... And this crash happened only once so far.

The ```invalid image data``` can be misleading here because i don't know how ```crashlytics``` is registering Keys, but i am having this error quite often in "non-fatal" errors especially when ```Image.network``` is unable to load a remote URL.

**Comment by Tom3652** [comment]
I have got a very similar crash yesterday : 

```
Fatal Exception: NSInvalidArgumentException
-[FChildrenNode encodeSuccessEnvelope:]: unrecognized selector sent to instance 0x281a2cec0
``` 

![Capture d’écran 2022-01-07 à 16 04 15](https://user-images.githubusercontent.com/32631467/148562858-495fe582-bdfc-4e15-b8aa-4e6d917254d3.png)

There is no link to ```photo_manager``` here however...  
Here is the full trace, tell me if you prefer i make a new issue : 

```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x9904c __exceptionPreprocess
1  libobjc.A.dylib                0x15f54 objc_exception_throw
2  CoreFoundation                 0x176014 +[NSObject(NSObject) _copyDescription]
3  CoreFoundation                 0x2e474 ___forwarding___
4  CoreFoundation                 0x2d5b0 _CF_forwarding_prep_0
5  Flutter                        0x4f02b0 (Missing)
6  Flutter                        0x3c664 (Missing)
7  Flutter                        0x3bc204 (Missing)
8  Flutter                        0x2eb0d8 (Missing)
9  Flutter                        0x2ee6a4 (Missing)
10 CoreFoundation                 0xae268 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
11 CoreFoundation                 0x32c40 __CFRunLoopDoTimer
12 CoreFoundation                 0x2d43c __CFRunLoopDoTimers
13 CoreFoundation                 0xbc58 __CFRunLoopRun
14 CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific
15 GraphicsServices               0x138c GSEventRunModal
16 UIKitCore                      0x5196a8 -[UIApplication _run]
17 UIKitCore                      0x2987f4 UIApplicationMain
18 Runner                         0x7ab0 main + 7 (AppDelegate.swift:7)
19 ???                            0x104ecda24 (Missing)

Crashed: com.google.firebase.crashlytics.ios.exception
0  FirebaseCrashlytics            0x1f44c FIRCLSProcessRecordAllThreads + 184
1  FirebaseCrashlytics            0x1f82c FIRCLSProcessRecordAllThreads + 1176
2  FirebaseCrashlytics            0x167f0 FIRCLSHandler + 48
3  FirebaseCrashlytics            0x10c78 __FIRCLSExceptionRecord_block_invoke + 92
4  libdispatch.dylib              0x4660 _dispatch_client_callout + 20
5  libdispatch.dylib              0x138b4 _dispatch_lane_barrier_sync_invoke_and_complete + 56
6  FirebaseCrashlytics            0x10414 FIRCLSExceptionRecord + 216
7  FirebaseCrashlytics            0x10720 FIRCLSExceptionRecordNSException + 560
8  FirebaseCrashlytics            0xffec FIRCLSTerminateHandler() + 404
9  libc++abi.dylib                0x12f58 std::__terminate(void (*)()) + 20
10 libc++abi.dylib                0x15efc __cxa_rethrow + 148
11 libobjc.A.dylib                0x17e8c objc_exception_rethrow + 44
12 CoreFoundation                 0x1f4a4 CFRunLoopRunSpecific + 836
13 GraphicsServices               0x138c GSEventRunModal + 164
14 UIKitCore                      0x5196a8 -[UIApplication _run] + 1100
15 UIKitCore                      0x2987f4 UIApplicationMain + 2092
16 Runner                         0x7ab0 main + 7 (AppDelegate.swift:7)
17 ???                            0x104ecda24 (Missing)

Thread
0  libsystem_kernel.dylib         0x1b2c __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1140 _pthread_wqthread + 364
2  libsystem_pthread.dylib        0xe94 start_wqthread + 8

com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Foundation                     0x18354 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 236
6  Foundation                     0x59c28 -[NSRunLoop(NSRunLoop) runUntilDate:] + 92
7  UIKitCore                      0x4928a4 -[UIEventFetcher threadMain] + 524
8  Foundation                     0x6836c __NSThread__start__ + 808
9  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
10 libsystem_pthread.dylib        0xea0 thread_start + 8

Thread
0  libsystem_kernel.dylib         0x1b2c __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1140 _pthread_wqthread + 364
2  libsystem_pthread.dylib        0xe94 start_wqthread + 8

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  CFNetwork                      0x27a6a0 _CFURLStorageSessionCopyIdentifier + 59784
6  Foundation                     0x6836c __NSThread__start__ + 808
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

Thread
0  libsystem_kernel.dylib         0x1b2c __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1140 _pthread_wqthread + 364
2  libsystem_pthread.dylib        0xe94 start_wqthread + 8

Thread
0  libsystem_kernel.dylib         0x1b2c __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1140 _pthread_wqthread + 364
2  libsystem_pthread.dylib        0xe94 start_wqthread + 8

io.flutter.1.ui
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Flutter                        0x2ee570 (Missing)
6  Flutter                        0x2edd30 (Missing)
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

io.flutter.1.raster
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Flutter                        0x2ee570 (Missing)
6  Flutter                        0x2edd30 (Missing)
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

io.flutter.1.io
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Flutter                        0x2ee570 (Missing)
6  Flutter                        0x2edd30 (Missing)
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.1
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.2
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.3
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.4
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.5
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

io.worker.6
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8254 _pthread_cond_wait + 1228
2  libc++.1.dylib                 0x9ddc std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 28
3  Flutter                        0x2e8928 (Missing)
4  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
5  libsystem_pthread.dylib        0xea0 thread_start + 8

dart:io EventHandler
0  libsystem_kernel.dylib         0x2e7c kevent + 8
1  Flutter                        0x4b14dc (Missing)
2  Flutter                        0x4e04e8 (Missing)
3  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
4  libsystem_pthread.dylib        0xea0 thread_start + 8

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

Thread
0  libsystem_pthread.dylib        0xe8c start_wqthread + 258

com.google.firebase.crashlytics.MachExceptionServer
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  FirebaseCrashlytics            0x19140 FIRCLSMachExceptionServer + 104
3  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
4  libsystem_pthread.dylib        0xea0 thread_start + 8

AVAudioSession Notify Thread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  AudioSession                   0x6308 CADeprecated::GenericRunLoopThread::Entry(void*) + 164
6  AudioSession                   0xfd64 CADeprecated::CAPThread::Entry(CADeprecated::CAPThread*) + 92
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

com.apple.CoreMotion.MotionThread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  CoreFoundation                 0x9fa78 CFRunLoopRun + 64
6  CoreMotion                     0x13544 CLMotionActivity::isTypeInVehicle(CLMotionActivity::Type) + 21264
7  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
8  libsystem_pthread.dylib        0xea0 thread_start + 8

com.squareup.SocketRocket.NetworkThread
0  libsystem_kernel.dylib         0x1504 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x1b9c mach_msg + 76
2  CoreFoundation                 0x7688 __CFRunLoopServiceMachPort + 372
3  CoreFoundation                 0xb97c __CFRunLoopRun + 1212
4  CoreFoundation                 0x1f3b8 CFRunLoopRunSpecific + 600
5  Foundation                     0x18354 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 236
6  FirebaseDatabase               0x52f5c -[_FSRRunLoopThread main] + 252
7  Foundation                     0x6836c __NSThread__start__ + 808
8  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
9  libsystem_pthread.dylib        0xea0 thread_start + 8

com.apple.CFSocket.private
0  libsystem_kernel.dylib         0x21d0 select$DARWIN_EXTSN + 8
1  libsystem_kernel.dylib         0x21d0 __select + 8
2  CoreFoundation                 0xaa740 __CFSocketManager + 648
3  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
4  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8

DartWorker
0  libsystem_kernel.dylib         0x1f90 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x8280 _pthread_cond_wait + 1272
2  Flutter                        0x5a221c (Missing)
3  Flutter                        0x5e8634 (Missing)
4  Flutter                        0x5a1b44 (Missing)
5  libsystem_pthread.dylib        0x19a4 _pthread_start + 148
6  libsystem_pthread.dylib        0xea0 thread_start + 8
```

![Capture d’écran 2022-01-07 à 16 10 09](https://user-images.githubusercontent.com/32631467/148563645-6aeb76a6-74b5-48a7-ba1d-b10b8dd76df7.png)

Again it's coming from one of my beta tester (same device actually) and i have not encountered anything like this myself or during development...

**Comment by gaaclarke** [comment]
Thanks @Tom3652, this still isn't enough information to fix the problem unfortunately because we are dealing what looks like a memory problem (nondeterministic + unrecognized selector) in an environment where we are running plugins.

Here's what I'd do if this was my app:
1) Try to talk to the user that is seeing this problem, try to figure out what steps they took in the app before the exception.
2) Try to reproduce the issue using memory debugging tools like malloc scribble (more info [here](https://developer.apple.com/library/archive/documentation/Performance/Conceptual/ManagingMemory/Articles/MallocDebug.html))

Hopefully step 2 will give you an error that is closer to the source of the problem.  It's almost certain that the error you are seeing is the ramifications of an earlier memory error, you see what I'm saying?  That's why this isn't actionable yet.

**Comment by Tom3652** [comment]
Thanks for your answer @gaaclarke ! 
I did as you said and I think i have spotted the issue now thanks to your explanation regarding memory.  
I am going to make a new release tonight for my testers and see during the week if this kind of crash still happens with similar scenarios.

If I was right, i will explain here the problem because it's complicated enough :)

**Comment by Tom3652** [comment]
So i had not the crash since the update i am guessing that what i thought was the issue. 

I am using ```photo_manager``` to retrieve asset from the phone's gallery and load them in a ```GridView.builder```.  
However, there was a problem with iCloud files that were taking too much time to be loaded on the next page ```DetailPage``` when clicking on an asset thumbnail inside the ```GridView```. 
To prevent this, i was checking during the ```itemBuilder``` if the asset needed to be downloaded from iCloud and I was starting the download if yes... That was probably causing memory issue as it attached many processes or something else.  

Thanks again for your help and explanations, if it comes back later i will leave a comment but there is no reason i think unless i do again nonsense like this :)

**Comment by gaaclarke** [comment]
Nice to hear, take care @Tom3652.
